scalalib/src/mill/scalalib/TestModuleUtil.scala

```diff
@@ -43,46 +43,91 @@ private[scalalib] object TestModuleUtil {
       EnvVars.MILL_WORKSPACE_ROOT -> Task.workspace.toString
     )
 
-    def runTestRunnerSubprocess(selectors2: Seq[String], base: os.Path) = {
-      val outputPath = base / "out.json"
-      val testArgs = TestArgs(
-        framework = testFramework,
-        classpath = runClasspath.map(_.path),
-        arguments = args,
-        sysProps = props,
-        outputPath = outputPath,
-        colored = Task.log.colored,
-        testCp = testClasspath.map(_.path),
-        globSelectors = selectors2
-      )
+    def prepareTestSelectorFolder(selectors2: Seq[String], base: os.Path): os.Path = {
+      val selectorFolder = base / "selectors"
+      os.makeDir.all(selectorFolder)
+      selectors2.zipWithIndex.foreach { case (s, i) =>
+        os.write.over(selectorFolder / s"selector-$i", s)
```

Comment:
How about we name the files according to the class name `s` rather than the index `i`? If we need to debug things being able to `ls` to see the queued test classes would be much easier than having to dig into each file individually